sympy/ntheory/factor_.py

```diff
@@ -509,22 +509,56 @@ def perfect_power(n, candidates=None, big=True, factor=True):
             if pp:
                 pp = (n.func(1, pp[0]), pp[1])
         else:
-            pp = perfect_power(p)
-            if pp:
-                num, e = pp
-                pq = perfect_power(q, [e])
-                if pq:
-                    den, _ = pq
-                    pp = n.func(num, den), e
+            pp_num = perfect_power(p)
+            if not pp_num:
+                return False
+
+            pp_den = perfect_power(q)
+            if not pp_den:
+                return False
+
+            num_base, num_exp = pp_num
+            den_base, den_exp = pp_den
+
+            common_exp = math.gcd(num_exp, den_exp)
+            if common_exp <= 1:
+                return False
+
+            final_num = num_base ** (num_exp // common_exp)
+            final_den = den_base ** (den_exp // common_exp)
+
+            return n.func(final_num, final_den), common_exp
         return pp
 
+    if candidates is not None:
+        candidates = set(candidates)
     n = as_int(n)
     if n < 0:
-        pp = perfect_power(-n)
-        if pp:
+        if candidates is None:
+            pp = perfect_power(-n, big=True, factor=factor)
+            if not pp:
+                return False
+
             b, e = pp
-            if e % 2:
+            e2 = e & (-e)
+            b, e = b ** e2, e // e2
+
+            if e <= 1:
```

Comment:
I'm glad to see that you identified this case, too. It is tested by -4 in the tests.